Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -613,116 +616,152 @@ public OrderTicket ExerciseOption(Symbol optionSymbol, int quantity, bool asynch
         {
-            return Order(strategy, Math.Abs(quantity), orderProperties);
+            return Order(strategy, Math.Abs(quantity), asynchronous, tag, orderProperties);
         }
 
         /// <summary>
         /// Sell Option Strategy (alias of Order)
         /// </summary>
         /// <param name="strategy">Specification of the strategy to trade</param>
         /// <param name="quantity">Quantity of the strategy to trade</param>
+        /// <param name="asynchronous">Send the order asynchronously (false). Otherwise we'll block until it fills</param>
+        /// <param name="tag">String tag for the order (optional)</param>
         /// <param name="orderProperties">The order properties to use. Defaults to <see cref="DefaultOrderProperties"/></param>
         /// <returns>Sequence of order tickets</returns>
         [DocumentationAttribute(TradingAndOrders)]
-        public IEnumerable<OrderTicket> Sell(OptionStrategy strategy, int quantity, IOrderProperties orderProperties = null)
+        public IEnumerable<OrderTicket> Sell(OptionStrategy strategy, int quantity, bool asynchronous = false, string tag = "", IOrderProperties orderProperties = null)
         {
-            return Order(strategy, Math.Abs(quantity) * -1, orderProperties);
+            return Order(strategy, Math.Abs(quantity) * -1, asynchronous, tag, orderProperties);
         }
 
         /// <summary>
         ///  Issue an order/trade for buying/selling an option strategy
         /// </summary>
         /// <param name="strategy">Specification of the strategy to trade</param>
         /// <param name="quantity">Quantity of the strategy to trade</param>
+        /// <param name="asynchronous">Send the order asynchronously (false). Otherwise we'll block until it fills</param>
+        /// <param name="tag">String tag for the order (optional)</param>
         /// <param name="orderProperties">The order properties to use. Defaults to <see cref="DefaultOrderProperties"/></param>
         /// <returns>Sequence of order tickets</returns>
         [DocumentationAttribute(TradingAndOrders)]
-        public IEnumerable<OrderTicket> Order(OptionStrategy strategy, int quantity, IOrderProperties orderProperties = null)
+        public IEnumerable<OrderTicket> Order(OptionStrategy strategy, int quantity, bool asynchronous = false, string tag = "", IOrderProperties orderProperties = null)
         {
-            return GenerateOrders(strategy, quantity, orderProperties);
+            return GenerateOptionStrategyOrders(strategy, quantity, asynchronous, tag, orderProperties);
         }
 
-        private IEnumerable<OrderTicket> GenerateOrders(OptionStrategy strategy, int strategyQuantity, IOrderProperties orderProperties = null)
+        public IEnumerable<OrderTicket> ComboOrder(OrderType type, List<Leg> legs, int quantity, decimal? limitPrice, bool asynchronous = false, string tag = "", IOrderProperties orderProperties = null)
```

Comment:
I suggest splitting this method into `ComboMarketOrder` and `ComboLimitOrder` (and `ComboLegLimitOrder`?) and removing the `OrderType` parameter. It will follow the `MarketOrder` and `LimitOrder` patterns, and it's more user-friendly.

Should it return a `List<OrderTicket>` or an `OrderTicket[]` to be Python-friendly?